resources/src/amount.rs

```diff
@@ -15,6 +15,10 @@ pub struct Amount(i64);
 impl Amount {
     /// Create an amount newtype
     pub fn new(amount: i64) -> Amount {
+        assert!(
+            amount >= 0,
+            format!("Amount must be greater than zero, got {}.", amount)
```

Comment:
this will always create the string even if we aren't panicking. What does the err say if you don't put the message in?